tests/FSharpx.Collections.Tests/PersistentVectorTest.fs

```diff
@@ -235,11 +235,21 @@ module PersistentVectorTests =
 
                 let vector2 = PersistentVector.map (fun x -> x * 2) (!vector)
 
-                let a = vector2 |> Seq.toArray 
+                let a = vector2 |> Seq.toArray
                 for i in 1..300 do i * 2 |> Expect.equal "map" a.[i-1] }
 
+            test "vector should allow mapi" {
+                let vector = ref PersistentVector.empty
```

Comment:
Why is ref cell needed? Can't we handle it using `mutable`?

Reply:
Copied from the test above "vector with 300 elements should allow pop"

Reply:
Happy to change it to `mutable` (I prefer that too) but was trying to follow style of existing code.

Reply:
I missed that the rest of the file approaches it like that. I would leave it as it is and potentially tackle all of the ref cells in tests in separate PR. @jackfoxy what's your opinion on that?

Reply:
OK to merge?